recipe/2023-11-validate-template-path.md

```diff
@@ -0,0 +1,96 @@
+# Validation for Terraform Recipe Template Paths
+
+* **Status**: Pending
+* **Author**: Shruthi Kumar (@sk593)
+
+## Overview
+
+Currently, we support only Terraform registry and HTTP URLs as allowed module sources for Terraform recipe template paths. However, we lack a proper validation mechanism for when an unsupported source is given during either recipe registration or recipe deployment. Consequently, when a deployment fails, the reason provided to users is ambiguous resulting into a poor user experience.
+
+## Terms and definitions
+
+| Term     | Definition                                                                                                                                                                                                 |
+| -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Terraform module source | The source argument in a module block tells Terraform where to find the source code for the desired child module |
+| Module registry | A module registry is the native way of distributing Terraform modules for use across multiple configurations |
+| Terraform registry | Terraform Registry is an index of modules shared publicly |
+| HTTP URLs | When you use an HTTP or HTTPS URL, Terraform will make a GET request to the given URL, which can return another source address | 
+
+## Objectives
+
+> **Issue Reference:** https://github.com/radius-project/radius/issues/6642
+
+### Goals
+
+- If a supported module source for a Terraform recipe is provided, the recipe registration should succeed 
+- If an unsupported module source for a Terraform recipe is provided, the recipe registration should fail and return a detailed error message to the user.
+
+
+### Non goals
+
+- Adding support and testing for additional module sources (Github, S3, etc). These will be addressed as needed based on customer feedback.  
+
+### User scenarios (optional)
+
+#### User story 1
+
+As a Radius user, I want to define a Terraform recipe with a Terraform module template path. My module source is a Terraform registry or HTTP URL. I am able to register my Terraform recipe with this source and deploy it successfully. 
+
+#### User story 2
+
+As a Radius user, I want to define a Terraform recipe with a Terraform module template path. My module source is not the supported Terraform registry or HTTP URLs (i.e. Github, local paths, S3 buckets, etc). When I try to register the recipe to my environment, I recieve an error message notifying me that I can only provide a Terraform module registry or HTTP URL as a valid module source using Radius. 
+
+## Design
+
+### Design details
+
+### API design (if applicable)
+N/A. We are updating an existing API to include more validation, but this won't include any public-facing or user-facing changes. 
```

Comment:
minor nitpick: tightening validation is **usually** a breaking change. I think we're ok with a breaking change in this case.